src/Val.cc

```diff
@@ -2810,6 +2809,12 @@ RecordVal::RecordVal(RecordTypePtr t, bool init_fields) : Val(t), is_managed(t->
         record_val.reserve(n);
 }
 
+RecordVal::RecordVal(RecordTypePtr t, const std::vector<std::optional<ZVal>>& init_vals)
```

Comment:
I seem to remember @bbannier saying elsewhere the compiler can do smarter things taking a value:
```suggestion
RecordVal::RecordVal(RecordTypePtr t, const std::vector<std::optional<ZVal>> init_vals)
```

Followed by a `std::move(init_vals)` on the assignment. And modifying the caller to use `std::move(...)`, too, if possible.

Reply:
Thanks for the comment. The caller can't do a move as it reuses the same vector for subsequent calls. Given that, is it still worth making this change? It seems either way there has to be a copy of the vector, and with the current way it's clear that there will just be one regardless ...

Reply:
I see your point. I'm fine with switching to accommodate that common idiom.